src/main/java/org/apache/flink/graph/streaming/library/ConnectedComponents.java

```diff
@@ -0,0 +1,129 @@
+
+	private long mergeWindowTime;
+
+	public ConnectedComponents(long mergeWindowTime) {
+		super(new UpdateCC(), new CombineCC(), new DisjointSet<K>(), mergeWindowTime, false);
+	}
+
+	/**
+	 * Implements EdgesFold Interface, applies a function to a vertex neighborhood
+	 * in the {@link GraphWindowStream#foldNeighbors(Object, EdgesFold)} method.
+	 *
+	 * @param <K> the vertex ID type
+	 */
+	public final static class UpdateCC<K extends Serializable> implements EdgesFold<K, NullValue, DisjointSet<K>> {
+
+		/**
+		 * Combines two edge values into one value of the same type.
+		 * The foldEdges function is consecutively applied to all edges of a neighborhood,
+		 * until only a single value remains.
+		 *
+		 * @param ds        the initial value and accumulator
+		 * @param vertex    the vertex ID
+		 * @param vertex2   the neighbor's ID
+		 * @param edgeValue the edge value
+		 * @return The data stream that is the result of applying the foldEdges function to the graph window.
+		 * @throws Exception
+		 */
+		@Override
+		public DisjointSet<K> foldEdges(DisjointSet<K> ds, K vertex, K vertex2, NullValue edgeValue) throws Exception {
+			ds.union(vertex, vertex2);
+			return ds;
+		}
+	}
+
+	/**
+	 * Reduce functions combine groups of elements to
+	 * a single value, by taking always two elements and combining them into one. Reduce functions
+	 * may be used on entire data sets, or on grouped data sets. In the latter case, each group is reduced
+	 * individually.
+	 * <p>
+	 * For a reduce functions that work on an entire group at the same time (such as the
+	 * MapReduce/Hadoop-style reduce), see {@link GroupReduceFunction}. In the general case,
+	 * ReduceFunctions are considered faster, because they allow the system to use more efficient
+	 * execution strategies.
+	 * <p>
+	 * The basic syntax for using a grouped ReduceFunction is as follows:
+	 * <pre>{@code
+	 * DataSet<X> input = ...;
+	 *
+	 * DataSet<X> result = input.groupBy(<key-definition>).reduce(new MyReduceFunction());
+	 * }</pre>
+	 * <p>
+	 * Like all functions, the ReduceFunction needs to be serializable, as defined in {@link java.io.Serializable}.
+	 */
+	public static class CombineCC<K extends Serializable> implements ReduceFunction<DisjointSet<K>> {
+
+		/**
+		 * The core method of ReduceFunction, combining two values into one value of the same type.
+		 * The reduce function is consecutively applied to all values of a group until only a single value remains.
+		 *
+		 * @param s1 The first value to combine.
+		 * @param s2 The second value to combine.
+		 * @return The combined value of both input values.
+		 * @throws Exception This method may throw exceptions. Throwing an exception will cause the operation
+		 *                   to fail and may trigger recovery.
+		 */
+		@Override
+		public DisjointSet<K> reduce(DisjointSet<K> s1, DisjointSet<K> s2) throws Exception {
```

Comment:
changed to:
Implements reduce method of ReduceFunction interface.
Two values of DisjointSet class are combined into one using merge method
of the DisjointSet class.
 In this case the merge method takes Connected Components values from different
partitions and merges them into one.